### PR TITLE
[Integration][Datadog] Datadog Teams and Users

### PR DESCRIPTION
**Pull request by shariff-6**
# Description

What -
Added support for Datadog Teams and Users in the integration, including fetching, processing, and enriching data related to teams and users.

Why -
Enable users to associate monitors, SLOs, and other Datadog resources with specific teams and users

How -

- Updated `resources/blueprints.json` to define schemas for Datadog Teams and Users.  
- Modified `client.py` to include methods for fetching teams and users, as well as members of specific teams.  
- Added `main.py` logic to enrich team data with associated members.  
- Updated `resources/port-app-config.yaml` to map Datadog Team and User properties to corresponding Port entities.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/datadog/.port/resources/port-app-config.yaml [hunk 1]
```diff
@@ -58,6 +93,13 @@ resources:
             owners: >-
               [.attributes.schema.contacts[] | select(.type == "email") |
               .contact]
+          relations:
+            team:
+              combinator: '"and"'
+              rules:
+                - property: '"handle"'
+                  operator: '"contains"'
```
Comment (PeyGis) [inline integrations/datadog/.port/resources/port-app-config.yaml:101]: instead of contains why not "="?

File: integrations/datadog/client.py [hunk 2]
```diff
@@ -158,6 +158,75 @@ async def _fetch_with_rate_limit_handling(
...
+                    "page[size]": page_size,
+                    "page[number]": page,
+                },
+            )
+
+            teams = result.get("data", [])
+            if not teams:
+                break
+
+            logger.info(f"Retrieved {len(teams)} teams")
```
Comment (PeyGis) [inline integrations/datadog/client.py:203]: let's remove this as we've doing the same in the main.py

**Review by PeyGis — CHANGES_REQUESTED** [review]
looks great. left some minor comments

**Review by phalbert — CHANGES_REQUESTED** [review]
A few comments @shariff-6

File: integrations/datadog/.port/resources/blueprints.json [hunk 3]
```diff
@@ -252,20 +355,7 @@
     },
     "mirrorProperties": {},
     "calculationProperties": {},
-    "aggregationProperties": {
```
[the remark this answers is not in the document]
Comment (shariff-6) [reply integrations/datadog/.port/resources/blueprints.json:255]: It was depending on an undefined blueprint, and failing on set 
datadogSloHistory, does this blueprint exist?

File: integrations/datadog/.port/spec.yaml [hunk 4]
```diff
@@ -11,6 +11,9 @@ features:
       - kind: slo
       - kind: sloHistory
       - kind: serviceMetric
+      - kind: team
```
[the remark this answers is not in the document]
Comment (shariff-6) [reply integrations/datadog/.port/spec.yaml]: :+1:

File: integrations/datadog/overrides.py [hunk 5]
```diff
@@ -65,9 +65,24 @@ class DatadogResourceConfig(ResourceConfig):
...
 
+class TeamSelector(Selector):
+    include_members: bool = Field(
+        alias="includeMembers",
+        default=False,
+        description="Whether to include the members of the team, defaults to false",
+    )
+
+
+class TeamResourceConfig(ResourceConfig):
```
[the remark this answers is not in the document]
Comment (shariff-6) [reply integrations/datadog/overrides.py:76]: :+1:

**Review by phalbert — APPROVED** [review]
LGTM

File: integrations/datadog/CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+## 0.1.68 (2024-12-23)
```
Comment (phalbert) [inline integrations/datadog/CHANGELOG.md:10]:
```suggestion
## 0.1.68 (2024-12-24)
```

File: integrations/datadog/main.py [hunk 7]
```diff
@@ -28,6 +35,28 @@ def init_client() -> DatadogClient:
...
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    dd_client = init_client()
+
+    selector = cast(TeamResourceConfig, event.resource_config).selector
+
+    async for teams in dd_client.get_teams():
+        logger.info(f"Received teams batch with {len(teams)} teams")
+        if selector.include_members:
+            teams = await dd_client.enrich_teams_with_members(teams)
+        yield teams
```
Comment (Tankilevitch) [inline integrations/datadog/main.py:48]: I think the `enrich_teams_with_members` should be just `get_team_members` and the logic of the merge will be in the main file

Comment (shariff-6) [reply integrations/datadog/main.py:48]: Okay, makes sense. 👍 .

**Review by Tankilevitch — APPROVED** [review]
LGTM, great job @shariff-6

**Pull request merged**